src/components/slideshow/slideshow.js

```diff
@@ -148,112 +179,134 @@ define(['dialogHelper', 'inputManager', 'connectionManager', 'layoutManager', 'f
                 setUserScalable(false);
-                stopInterval();
             });
 
             inputManager.on(window, onInputCommand);
             document.addEventListener((window.PointerEvent ? 'pointermove' : 'mousemove'), onPointerMove);
 
-            dlg.addEventListener('close', onDialogClosed);
+            dialog.addEventListener('close', onDialogClosed);
 
             if (options.interactive) {
-                loadSwiper(dlg);
+                loadSwiper(dialog, options);
             }
         }
 
+        /**
+         * Handles OSD changes when the autoplay is started.
+         */
         function onAutoplayStart() {
-            var btnSlideshowPause = dlg.querySelector('.btnSlideshowPause i');
+            var btnSlideshowPause = dialog.querySelector('.btnSlideshowPause i');
             if (btnSlideshowPause) {
-                btnSlideshowPause.classList.remove("play_arrow");
-                btnSlideshowPause.classList.add("pause");
+                btnSlideshowPause.classList.replace("play_arrow", "pause");
             }
         }
 
+        /**
+         * Handles OSD changes when the autoplay is stopped.
+         */
         function onAutoplayStop() {
-            var btnSlideshowPause = dlg.querySelector('.btnSlideshowPause i');
+            var btnSlideshowPause = dialog.querySelector('.btnSlideshowPause i');
             if (btnSlideshowPause) {
-                btnSlideshowPause.classList.remove("pause");
-                btnSlideshowPause.classList.add("play_arrow");
+                btnSlideshowPause.classList.replace("pause", "play_arrow");
             }
         }
 
-        function loadSwiper(dlg) {
-
+        /**
+         * Initializes the Swiper instance and binds the relevant events.
+         * @param {HTMLElement} dialog - Element containing the dialog.
+         * @param {Object} options - Options used to initialize the Swiper instance.
+         */
+        function loadSwiper(dialog, options) {
+            var slides;
             if (currentOptions.slides) {
-                dlg.querySelector('.swiper-wrapper').innerHTML = currentOptions.slides.map(getSwiperSlideHtmlFromSlide).join('');
+                slides = currentOptions.slides;
             } else {
-                dlg.querySelector('.swiper-wrapper').innerHTML = currentOptions.items.map(getSwiperSlideHtmlFromItem).join('');
+                slides = currentOptions.items;
             }
 
             require(['swiper'], function (Swiper) {
-
-                swiperInstance = new Swiper(dlg.querySelector('.slideshowSwiperContainer'), {
+                swiperInstance = new Swiper(dialog.querySelector('.slideshowSwiperContainer'), {
                     // Optional parameters
                     direction: 'horizontal',
-                    loop: options.loop !== false,
-                    autoplay: {
-                        delay: options.interval || 8000
+                    loop: false,
```

Comment:
why disable looping? some slideshows could be used as looped

Reply:
In order to pre-load the next and previous slides (for a nicer experience), I enabled Swiper's "virtual slides".

This essentially keeps X slides rendered and removes/adds slides as the user navigates. This also helps by not loading all the DOM structure of the slides in one go.

However, in the current Swiper version, you can't enable looping with this (per their documentation).

I agree that looping would be useful, but I think the virtual slides things makes a sizeable difference to how nice the slideshow is to use.

Reply:
add a comment explaining this to the code then, please